gk/main.c

```diff
@@ -1123,6 +1327,13 @@ process_pkts_front(uint16_t port_front, uint16_t port_back,
 					continue;
 				}
 
+				if ((ret = update_ip_hop_count(&packet,
+						num_pkts, icmp_bufs)) < 0) {
+					if (ret != -ETIMEDOUT)
+						drop_packet(pkt);
```

Comment:
This code is error-prone, make `update_ip_hop_count()` always free the packet if it's not to be forwarded, so callers don't need to differentiate errors. This change will affect all callers of `update_ip_hop_count()`.